src/acars_router_servers/sender_servers.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Mike Nye, Fred Clausen
+//
+// Licensed under the MIT license: https://opensource.org/licenses/MIT
+// Permission is granted to use, copy, modify, and redistribute the work.
+// Full license information available in the project LICENSE file.
+//
+
+use crate::generics::{reconnect_options, SenderServer, SenderServerConfig, Shared};
+use crate::helper_functions::should_start_service;
+use crate::tcp_serve_server::TCPServeServer;
+use crate::udp_sender_server::UDPSenderServer;
+use log::{debug, error};
+use serde_json::Value;
+use std::sync::Arc;
+use stubborn_io::StubbornTcpStream;
+use tmq::{publish, Context};
+use tokio::net::{TcpListener, UdpSocket};
+use tokio::sync::mpsc::{Receiver, Sender};
+use tokio::sync::{mpsc, Mutex};
+
+pub async fn start_sender_servers(
+    config: &SenderServerConfig,
+    rx_processed: Receiver<Value>,
+    server_type: String,
+) {
+    // Flow is check and see if there are any configured outputs for the queue
+    // If so, start it up and save the transmit channel to the list of sender servers.
+    // Then start watchers for the input queue
+
+    let sender_servers: Arc<Mutex<Vec<Sender<Value>>>> = Arc::new(Mutex::new(Vec::new()));
+
+    if should_start_service(config.send_udp()) {
+        // Start the UDP sender servers for {server_type}
+
+        match UdpSocket::bind("0.0.0.0:0".to_string()).await {
+            Ok(socket) => {
+                let (tx_processed, rx_processed) = mpsc::channel(32);
+                let udp_sender_server = UDPSenderServer {
+                    host: config.send_udp().clone(),
+                    proto_name: server_type.clone(),
+                    socket: socket,
+                    max_udp_packet_size: config.max_udp_packet_size,
+                    channel: rx_processed,
+                };
+
+                let new_state = Arc::clone(&sender_servers);
+                new_state.lock().await.push(tx_processed.clone());
+
+                tokio::spawn(async move {
+                    udp_sender_server.send_message().await;
+                });
+            }
+            Err(e) => {
+                error!("[{}] Failed to start UDP sender server: {}", server_type, e);
+            }
+        };
+    }
+
+    if should_start_service(config.send_tcp()) {
+        for host in config.send_tcp() {
+            let new_state = Arc::clone(&sender_servers);
+            let hostname = host.clone();
+            let s_type = server_type.clone();
+            tokio::spawn(async move { start_tcp(hostname, s_type, new_state).await });
+        }
+    }
+
+    if should_start_service(config.serve_tcp()) {
+        // Start the TCP servers for {server_type}
+
+        for host in config.serve_tcp() {
+            let hostname = "0.0.0.0:".to_string() + host.as_str();
```

Comment:
Hi there,

Sorry to be a pain - this doesn't look right (either host is badly named - port - or your gonna get some weird hostnames

Reply:
Thanks for taking a look @bluntelk. @fredclausen FYI this is "Boxie" the discord.

Reply:
I need my morning coffee but I think this is just a poorly named variable. The "host" in this instance is storing only port values, and it is checked during initial load up before it gets to this code to ensure it (and the other user input) are all valid.

Reply:
Actually that is a lie, at least the second half. There are a few input values I added that I forgot to make sure the input is valid and sanitized.

Reply:
If you're joining two str values like that, probably want something more like `let hostname: String = format!("0.0.0.0:{}", host.as_str());`

The `format!()` macro is fantastic for doing things like formatting of values.

Reply:
> If you're joining two str values like that, probably want something more like `let hostname: String = format!("0.0.0.0:{}", host.as_str());`
> 
> The `format!()` macro is fantastic for doing things like formatting of values.

I'm using the format macro now, thank you for that. This is my first go at writing Rust and I kind of just jumped in head first so obvious things like that I'm not surprised I missed.